src/snabbkaffe.erl

```diff
@@ -106,6 +113,30 @@
 %% API functions
 %%====================================================================
 
+-spec tp(term(), logger:level(), kind(), map()) -> ok.
+-ifndef(CONCUERROR).
+tp(Location, Level, Kind, Data) ->
+  Fun = persistent_term:get(snabbkaffe_tp_fun),
+  apply(Fun, [Location, Level, Kind, Data]).
+-else.
+tp(Location, Level, Kind, Data) ->
+  local_tp(Location, Level, Kind, Data).
+-endif. %% CONCUERROR
+
+-spec local_tp(term(), logger:level(), kind(), map()) -> ok.
+local_tp(Location, _Level, Kind, Data) ->
+  Event = Data #{?snk_kind => Kind},
+  snabbkaffe_nemesis:maybe_crash(Location, Event),
+  snabbkaffe_collector:tp(Event).
+
+-spec remote_tp(term(), logger:level(), kind(), map()) -> ok.
+remote_tp(Location, Level, Kind, Data) ->
+  Node = persistent_term:get(snabbkaffe_remote),
+  case rpc:call(Node, snabbkaffe, tp, [Location, Level, Kind, Data], infinity) of
```

Comment:
That's a tricky question... I believe this implementation should be sufficient, because it blocks process producing the event until the event is recorded in the collector.

Reply:
P.S. I checked this with concuerror, and it seems like it didn't find any race conditions with proxy rpc process. I won't add this testcase to the suite, because it required some ugly hacks, and concuerror is not very reliable when it comes to erlang distribution.